### PR TITLE
chore: add BPDM to umbrella chart

### PR DESCRIPTION
**Pull request by gomezbc**
## Description

This PR adds BPDM chart to the umbrella repository. Due to incompatible Portal version, I had to bump portal chart to 2.0.0. As mentioned in this issue, https://github.com/eclipse-tractusx/portal/issues/248, in version 1.8.1 of Portal, it calls this enpoint `/companies/test-company/v6/` of the gate, which is incompatible with the added gate.

This BPDM is prepared to work with centralidp, but as it is planned for future release https://github.com/eclipse-tractusx/sig-release/issues/751, currently it will return 403 error in BPN step.

https://github.com/eclipse-tractusx/sig-release/issues/710

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

**Comment by gomezbc** [comment]
@evegufy

File: charts/umbrella/Chart.yaml
Comment (evegufy) [inline charts/umbrella/Chart.yaml:39]: Could you please restrict your changes to bpdm?
I know it won't work as a whole as long as portal isn't upgrade as well  (that's why we work on a separate upgrade branch until all components are ready). I'd upgrade portal, centraldip and sharedidp all together in a separate PR.

**Review by evegufy — CHANGES_REQUESTED** [review]
Hi @gomezbc thank you so much for the contribution!

File: charts/umbrella/Chart.yaml
Comment (gomezbc) [reply charts/umbrella/Chart.yaml:39]: Understood. I'll revert portal version upgrade. I'm ready to help with the portal, centraldip, and sharedidp upgrades if needed.

File: charts/umbrella/Chart.yaml
Comment (evegufy) [inline charts/umbrella/Chart.yaml:75]: small typo in Business

File: charts/umbrella/values.yaml
Comment (evegufy) [inline charts/umbrella/values.yaml:685]: by default we add all the chart with enabled set to false

for testing we have 2 different values files, could you please add and enable bpdm in the one for the shared services? https://github.com/eclipse-tractusx/tractus-x-umbrella/blob/b6d50476ab4689b64e4b6418b538f9669407b9eb/charts/values-test-shared-services.yaml

**Review by evegufy — CHANGES_REQUESTED** [review]

**Review by evegufy — CHANGES_REQUESTED** [review]
could you pleas also raise the chart version to 0.17.0? (the helm test workflow requires the increment to run https://github.com/eclipse-tractusx/tractus-x-umbrella/actions/runs/9844462507/job/27182944379?pr=106)

File: charts/umbrella/Chart.yaml
Comment (gomezbc) [reply charts/umbrella/Chart.yaml]: Sure! In that case, should I create a separate pull request for each component, or do I upgrade them in one?

Comment (evegufy) [reply charts/umbrella/Chart.yaml]: I think one pull request with the upgrade of all 3 components would be also good :)

**Comment by evegufy** [comment]
@gomezbc the review is currently blocked by the merged of https://github.com/eclipse-tractusx/tractus-x-umbrella/pull/107, after it has been merged it would be great if you could rebase to the upgrade/24.05 branch

**Comment by evegufy** [comment]
> @gomezbc the review is currently blocked by the merged of #107, after it has been merged it would be great if you could rebase to the upgrade/24.05 branch

@gomezbc the pr #107 has been merged now, could you please rebase? then the helm test workflow should pass the check for chart version bump

**Comment by gomezbc** [comment]
@evegufy Done 👍🏼

**Review by evegufy — CHANGES_REQUESTED** [review]
Hi @gomezbc the helm test workflow did its job correctly now :) and uncovered some [linting issues,](https://github.com/eclipse-tractusx/tractus-x-umbrella/actions/runs/9870098659/job/27280181316?pr=106#step:13:58) could you please fix them?

**Comment by gomezbc** [comment]
@evegufy I have resolved the linting issues. I noticed that other components disable PostgreSQL persistence by default, so I have done the same for the BPDM database.

**Review by evegufy — CHANGES_REQUESTED** [review]
one linting issue still:
> charts/umbrella/values.yaml
>  Error: 748:19 [trailing-spaces] trailing spaces

**Review by evegufy — COMMENTED** [review]
Hi @gomezbc sorry for the delay, I finally came around to reviewing again.
When install I'm facing 401 errors for pool, gate and dummy:
[logs-from-bpdm-pool-in-umbrella-bpdm-pool-f5bb96f6d-d8xtn.log](https://github.com/user-attachments/files/16287944/logs-from-bpdm-pool-in-umbrella-bpdm-pool-f5bb96f6d-d8xtn.log)
[logs-from-bpdm-cleaning-service-dummy-in-umbrella-bpdm-cleaning-service-dummy-678cc46bb9-hps8t.log](https://github.com/user-attachments/files/16287965/logs-from-bpdm-cleaning-service-dummy-in-umbrella-bpdm-cleaning-service-dummy-678cc46bb9-hps8t.log)
[logs-from-bpdm-gate-in-umbrella-bpdm-gate-6d6cc7fdff-8mnkp.log](https://github.com/user-attachments/files/16287967/logs-from-bpdm-gate-in-umbrella-bpdm-gate-6d6cc7fdff-8mnkp.log)

This might be due to missing the bdpm config discussed here https://github.com/eclipse-tractusx/portal-backend/issues/819#issuecomment-2213538297

@nicoprow has opened https://github.com/eclipse-tractusx/tractus-x-umbrella/pull/108 which might contains the config.

**Review by evegufy — COMMENTED** [review]
Hi @gomezbc sorry for the delay, I finally came around to reviewing again.
When install I'm facing 401 errors for pool, gate and dummy:
[logs-from-bpdm-pool-in-umbrella-bpdm-pool-f5bb96f6d-d8xtn.log](https://github.com/user-attachments/files/16287944/logs-from-bpdm-pool-in-umbrella-bpdm-pool-f5bb96f6d-d8xtn.log)
[logs-from-bpdm-cleaning-service-dummy-in-umbrella-bpdm-cleaning-service-dummy-678cc46bb9-hps8t.log](https://github.com/user-attachments/files/16287965/logs-from-bpdm-cleaning-service-dummy-in-umbrella-bpdm-cleaning-service-dummy-678cc46bb9-hps8t.log)
[logs-from-bpdm-gate-in-umbrella-bpdm-gate-6d6cc7fdff-8mnkp.log](https://github.com/user-attachments/files/16287967/logs-from-bpdm-gate-in-umbrella-bpdm-gate-6d6cc7fdff-8mnkp.log)

This might be due to missing the bdpm config discussed here https://github.com/eclipse-tractusx/portal-backend/issues/819#issuecomment-2213538297

@nicoprow has opened https://github.com/eclipse-tractusx/tractus-x-umbrella/pull/108 which might contains the config.

**Comment by nicoprow** [comment]
> Hi @gomezbc sorry for the delay, I finally came around to reviewing again. When install I'm facing 401 errors for pool, gate and dummy: [logs-from-bpdm-pool-in-umbrella-bpdm-pool-f5bb96f6d-d8xtn.log](https://github.com/user-attachments/files/16287944/logs-from-bpdm-pool-in-umbrella-bpdm-pool-f5bb96f6d-d8xtn.log) [logs-from-bpdm-cleaning-service-dummy-in-umbrella-bpdm-cleaning-service-dummy-678cc46bb9-hps8t.log](https://github.com/user-attachments/files/16287965/logs-from-bpdm-cleaning-service-dummy-in-umbrella-bpdm-cleaning-service-dummy-678cc46bb9-hps8t.log) [logs-from-bpdm-gate-in-umbrella-bpdm-gate-6d6cc7fdff-8mnkp.log](https://github.com/user-attachments/files/16287967/logs-from-bpdm-gate-in-umbrella-bpdm-gate-6d6cc7fdff-8mnkp.log)
> 
> This might be due to missing the bdpm config discussed here [eclipse-tractusx/portal-backend#819 (comment)](https://github.com/eclipse-tractusx/portal-backend/issues/819#issuecomment-2213538297)
> 
> @nicoprow has opened #108 which might contains the config.

@gomezbc:

As @evegufy already said, you can use the newest hotfix version for BPDM 5.0.3 with the configuration provided in #108. The files also contain README instructions and a workflow to test the BPDM chart dependency setup.

The new version has the advantage that the postgres does not need to have a fullnameOverride anymore. 

The ingress can be changed for BPDM but I experienced compatibility problems with the Portal. At least for the Partner Network page, I discovered that that Portal version has outdated endpoint URLs when trying to access the Pool. I don't see a way to fix that except upgrading the Portal so the correct endpoint paths are used.

I did not try the registration workflow with the Portal, so I don't know which path to the BPDM Gate the Portal expects there.

File: charts/umbrella/README.md
Comment (nicoprow) [inline charts/umbrella/README.md:129]: The cleaning service does not have a API and does not need to be exposed. Have a look here to see the recommended settings: #108

File: charts/umbrella/values.yaml
Comment (nicoprow) [inline charts/umbrella/values.yaml:714]: not needed anymore, the default configuration already enbales auth. see: https://github.com/eclipse-tractusx/bpdm/blob/release/6.0.x/INSTALL.md

File: charts/umbrella/values.yaml
Comment (nicoprow) [inline charts/umbrella/values.yaml:684]: bpdm-gate, bpdm-pool and bpdm-orchestrator expect default permissions that are different from the Central-IDP version in the umbrella chart. Therefore the default permissions have to be overwritten like seen here #108

**Comment by gomezbc** [comment]
Hi @nicoprow, thanks for the comments. I have review your configuration and is awesome. As you have a better understanding of BPDM, I think the best approach would be to use your configuration.

Regarding the compatibility with the portal, in my local setup, I had to upgrade to version 2.0.0 to make it compatible with BPDM. Using that version, the registration process works correctly.

As I discussed with @evegufy (https://github.com/eclipse-tractusx/tractus-x-umbrella/pull/106#discussion_r1669237002), I will provide an upgrade to the portal, centralidp, and sharedidp. I will try to provide it as soon as I can.

**Comment by gomezbc** [comment]
Hi @nicoprow

After creating the PR for the portal version update https://github.com/eclipse-tractusx/tractus-x-umbrella/pull/110, I switched my configuration to yours to connect the portal with BPDM. However, I encountered some permission issues. Upon comparing the roles required by BPDM with those defined in CentralIdp, I resolved these issues.

The roles you defined for the Gate, Pool, and Orchestrator do not match those used by the CentralIdp sa-cl7-cx-5 client.

For example, here are the roles you set for Gate (Cl16-CX-BPDMGate):
```yaml
permissions:
  readInputPartner: view_company_data
  writeInputPartner: update_company_data
  readOutputPartner: view_shared_data
  readInputChangelog: view_company_data
  readOutputChangelog: view_shared_data
  readSharingState: view_company_data
  writeSharingState: update_company_data
  read_stats: view_company_data
```
And here are the roles provided by CentralIdp:

image

To fix the Gate and Pool permissions, I deleted those settings, as the default configuration matches Cl16-CX-BPDMGate and Cl17-CX-BPDM.

For the Orchestrator, I modified the values you set in permissions to these:
```yaml
permissions:
  createTask: "write_partner"
  readTask: "write_partner"
  reservation:
    clean: "write_partner"
    cleanAndSync: "write_partner"
    poolSync: "write_partner"
  result:
    clean: "write_partner"
    cleanAndSync: "write_partner"
    poolSync: "write_partner"
```

> [!warning]
> I have successfully completed the BPN creation process from the portal using the updated portal version [here](https://github.com/eclipse-tractusx/tractus-x-umbrella/pull/110). It will not be possible to complete the BPN creation process with the portal version defined in this PR.

**Comment by evegufy** [comment]
> > Hi @gomezbc sorry for the delay, I finally came around to reviewing again. When install I'm facing 401 errors for pool, gate and dummy: [logs-from-bpdm-pool-in-umbrella-bpdm-pool-f5bb96f6d-d8xtn.log](https://github.com/user-attachments/files/16287944/logs-from-bpdm-pool-in-umbrella-bpdm-pool-f5bb96f6d-d8xtn.log) [logs-from-bpdm-cleaning-service-dummy-in-umbrella-bpdm-cleaning-service-dummy-678cc46bb9-hps8t.log](https://github.com/user-attachments/files/16287965/logs-from-bpdm-cleaning-service-dummy-in-umbrella-bpdm-cleaning-service-dummy-678cc46bb9-hps8t.log) [logs-from-bpdm-gate-in-umbrella-bpdm-gate-6d6cc7fdff-8mnkp.log](https://github.com/user-attachments/files/16287967/logs-from-bpdm-gate-in-umbrella-bpdm-gate-6d6cc7fdff-8mnkp.log)
> > This might be due to missing the bdpm config discussed here [eclipse-tractusx/portal-backend#819 (comment)](https://github.com/eclipse-tractusx/portal-backend/issues/819#issuecomment-2213538297)
> > @nicoprow has opened #108 which might contains the config.
> 
> @gomezbc:
> 
> As @evegufy already said, you can use the newest hotfix version for BPDM 5.0.3 with the configuration provided in #108. The files also contain README instructions and a workflow to test the BPDM chart dependency setup.
> 
> The new version has the advantage that the postgres does not need to have a fullnameOverride anymore.
> 
> The ingress can be changed for BPDM but I experienced compatibility problems with the Portal. At least for the Partner Network page, I discovered that that Portal version has outdated endpoint URLs when trying to access the Pool. I don't see a way to fix that except upgrading the Portal so the correct endpoint paths are used.
> 
> I did not try the registration workflow with the Portal, so I don't know which path to the BPDM Gate the Portal expects there.

Just quickly clarifying on what you mentioned regarding outdated endpoints in the portal: I assume you tested with the version of the portal which is currently still in the main branch of the umbrella repo, which is the portal version 1.8.0 for the release 24.03, that one is expected to be not compatible with the BPDM version for 24.05, the portal version for the 24.05 release is the 2.0.0.
Overview of compatible version for the for R24.05 https://eclipse-tractusx.github.io/CHANGELOG#2405---2024-05-29

**Comment by nicoprow** [comment]
> Hi @nicoprow
> 
> After creating the PR for the portal version update #110, I switched my configuration to yours to connect the portal with BPDM. However, I encountered some permission issues. Upon comparing the roles required by BPDM with those defined in CentralIdp, I resolved these issues.
> 
> The roles you defined for the Gate, Pool, and Orchestrator do not match those used by the CentralIdp sa-cl7-cx-5 client.
> 
> For example, here are the roles you set for Gate (Cl16-CX-BPDMGate):
> 
> ```yaml
> permissions:
>   readInputPartner: view_company_data
>   writeInputPartner: update_company_data
>   readOutputPartner: view_shared_data
>   readInputChangelog: view_company_data
>   readOutputChangelog: view_shared_data
>   readSharingState: view_company_data
>   writeSharingState: update_company_data
>   read_stats: view_company_data
> ```
> 
> And here are the roles provided by CentralIdp:
> 
> image
> 
> To fix the Gate and Pool permissions, I deleted those settings, as the default configuration matches Cl16-CX-BPDMGate and Cl17-CX-BPDM.
> 
> For the Orchestrator, I modified the values you set in permissions to these:
> 
> ```yaml
> permissions:
>   createTask: "write_partner"
>   readTask: "write_partner"
>   reservation:
>     clean: "write_partner"
>     cleanAndSync: "write_partner"
>     poolSync: "write_partner"
>   result:
>     clean: "write_partner"
>     cleanAndSync: "write_partner"
>     poolSync: "write_partner"
> ```
> 
> Warning
> 
> I have successfully completed the BPN creation process from the portal using the updated portal version [here](https://github.com/eclipse-tractusx/tractus-x-umbrella/pull/110). It will not be possible to complete the BPN creation process with the portal version defined in this PR.

Yea, your permission configuration is exactly right to do for the Central-IDP 24.05 release. Great that you made it work!

If you want you can remove the base-URL overwrite for the BPDM clients as they are able to find each other by service name. I don't think is necessary to change it but I thought I should mention it (see the comment below)

File: charts/umbrella/values.yaml
Comment (nicoprow) [inline charts/umbrella/values.yaml:748]: On default BPDM services try to find each other by their release name in the cluster. Since the services are created by the same Helm release and are running in the same cluster/namespace we can remove the base-url overwrite here.

File: charts/umbrella/values.yaml
Comment (nicoprow) [inline charts/umbrella/values.yaml:752]: Same here

File: charts/umbrella/values.yaml
Comment (nicoprow) [inline charts/umbrella/values.yaml:804]: Same here

File: charts/umbrella/values.yaml
Comment (nicoprow) [inline charts/umbrella/values.yaml:866]: Same here

**Review by nicoprow — COMMENTED** [review]
Comments below as an option to further simplify the configuration.

Since there is now also the values-test-bpdm.yaml inside I would also expect the BPDM chart test to be part of this pull request.

**Comment by evegufy** [comment]
@evegufy will check out the chart testing

**Comment by evegufy** [comment]
Hi @gomezbc https://github.com/eclipse-tractusx/tractus-x-umbrella/pull/113 has been merged, could you please add bpdm to the the 1rst test file?

**Comment by gomezbc** [comment]
> Hi @gomezbc #113 has been merged, could you please add bpdm to the the 1rst test file?

Done! 😀

**Review by evegufy — APPROVED** [review]

**Comment by evegufy** [comment]
> > Hi @gomezbc #113 has been merged, could you please add bpdm to the the 1rst test file?
> 
> Done! 😀

Awesome :)

**Pull request merged**
